nexus/db-queries/src/db/datastore/region.rs

```diff
@@ -259,9 +260,12 @@ impl DataStore {
         let query = crate::db::queries::region_allocation::allocation_query(
             volume_id,
             maybe_snapshot_id,
-            block_size,
-            blocks_per_extent,
-            extent_count,
+            RegionParameters {
```

Comment:
This path is not used for snapshots, right?  This is only for RW regions?

Reply:
This part of the commit was pulled out of the snapshot replacement work, which _will_ call `arbitrary_region_allocate` to allocate read-only regions for snapshot volumes. Right now this is only for allocating RW regions.